### PR TITLE
Survive analyzer crash

### PR DESCRIPTION
**Pull request by sigurdm**
Fixes: https://github.com/dart-lang/pana/issues/827

### CODE REVIEW

File: lib/src/package_analyzer.dart [hunk 1]
```diff
@@ -142,11 +142,10 @@ class PackageAnalyzer {
     if (pkgResolution != null) {
       List<CodeProblem> analyzerItems;
       if (dartFiles.isNotEmpty) {
-        analyzerItems = await context.staticAnalysis();
-      } else {
-        analyzerItems = <CodeProblem>[];
```
Comment (isoos) [inline lib/src/package_analyzer.dart:147]: I think this part may be important, but may not be anymore....

Comment (sigurdm) [reply lib/src/package_analyzer.dart:147]: We have a null-check just below, and don't use the variable after that. I think it is all right.

File: lib/src/package_analyzer.dart [hunk 2]
```diff
@@ -142,11 +142,11 @@ class PackageAnalyzer {
...
+        try {
+          analyzerItems = await context.staticAnalysis();
+          // ignore: empty_catches
+        } on ToolException {}
```
Comment (isoos) [inline lib/src/package_analyzer.dart:148]: How about the following?
```
} on ToolException catch (e) {
  errors.add(messages.runningDartanalyzerFailed(usesFlutter, e.message));
}
```

**Review by isoos — APPROVED** [review]

**Review by isoos — APPROVED** [review]
Approval pending on the test fixes + changelog entry.

File: lib/src/package_analyzer.dart (hunk 2 above)
Comment (sigurdm) [reply lib/src/package_analyzer.dart]: Done

**Review by isoos — APPROVED** [review]

**Pull request merged**